src/extras/named-exports.js

```diff
@@ -42,10 +42,10 @@
           execute.call(this);
           // do a bulk export of the default export object
           // to export all its names as named exports
-          if (hasDefaultExport && typeof defaultExport === 'object')
+          if (hasDefaultExport)
             for (var exportName in defaultExport) {
               // default is not a named export
-              if (exportName !== 'default') {
+              if (defaultExport.hasOwnProperty(exportName) && exportName !== 'default') {
```

Comment:
I'm wondering if there is ever a case where the defaultExport does not have a `hasOwnProperty` function. The `defaultExport` can have any type of value. I've checked and numbers, strings, symbols, functions, objects, and arrays and they do have it. However, it's possible that someone could modify the prototype of the value such that it doesn't have a default export. An example is that ES module "namespace" objects do not have the object prototype, and are created with `Object.create(null)`.

A tangential example where the object prototype is removed from some things - https://github.com/systemjs/systemjs/issues/2147. That particular example doesn't demonstrate a specific case where the code here would break things, but just shows that there are situations where values do not have the Object prototype.

In such cases where hasOwnProperty does not exist, this code will throw an error. Perhaps the safer implementation would be this?

```suggestion
              if (Object.prototype.hasOwnProperty.call(defaultExport, exportName) && exportName !== 'default') {
```

Reply:
Yes, `Object.hasOwnProperty` is preferred for this reason.